airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/cursor_based/MssqlCursorBasedStateManager.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.source.mssql.cursor_based;
+
+import com.google.common.collect.Lists;
+import io.airbyte.cdk.integrations.source.relationaldb.CursorInfo;
+import io.airbyte.cdk.integrations.source.relationaldb.models.CursorBasedStatus;
+import io.airbyte.cdk.integrations.source.relationaldb.models.InternalModels.StateType;
+import io.airbyte.cdk.integrations.source.relationaldb.state.StreamStateManager;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.protocol.models.v0.AirbyteStateMessage;
+import io.airbyte.protocol.models.v0.AirbyteStateMessage.AirbyteStateType;
+import io.airbyte.protocol.models.v0.AirbyteStreamNameNamespacePair;
+import io.airbyte.protocol.models.v0.AirbyteStreamState;
+import io.airbyte.protocol.models.v0.ConfiguredAirbyteCatalog;
+import io.airbyte.protocol.models.v0.StreamDescriptor;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MssqlCursorBasedStateManager extends StreamStateManager {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(MssqlCursorBasedStateManager.class);
+
+  public MssqlCursorBasedStateManager(final List<AirbyteStateMessage> airbyteStateMessages, final ConfiguredAirbyteCatalog catalog) {
+    super(airbyteStateMessages, catalog);
+  }
+
+  @Override
+  public AirbyteStateMessage toState(final Optional<AirbyteStreamNameNamespacePair> pair) {
+    if (pair.isPresent()) {
+      final Map<AirbyteStreamNameNamespacePair, CursorInfo> pairToCursorInfoMap = getPairToCursorInfoMap();
+      final Optional<CursorInfo> cursorInfo = Optional.ofNullable(pairToCursorInfoMap.get(pair.get()));
+
+      if (cursorInfo.isPresent()) {
+        LOGGER.debug("Generating state message for {}...", pair);
+        return new AirbyteStateMessage()
+            .withType(AirbyteStateType.STREAM)
+            // Temporarily include legacy state for backwards compatibility with the platform
```

Comment:
Why do we need legacy state again?

Reply:
This was lifted from MySqlCursorBasedStateManager.
The idea I think is to not break existing saved legacy state already in db